akka-http2-support/src/main/scala/akka/http/impl/engine/http2/ResponseRendering.scala

```diff
@@ -41,4 +58,68 @@ private[http2] object ResponseRendering {
       response.entity.dataBytes
     )
   }
+
+  private def renderHeaders(
+    headersSeq:   immutable.Seq[HttpHeader],
+    headerPairs:  VectorBuilder[(String, String)],
+    serverHeader: Option[Server],
+    log:          LoggingAdapter
+  ): Unit = {
+    def suppressionWarning(h: HttpHeader, msg: String): Unit =
+      log.warning("Explicitly set HTTP header '{}' is ignored, {}", h, msg)
+
+    // optimized, as it is done for every response
+    val headers = headersSeq.toArray
```

Comment:
Should bench this, may be overdoing it (and may be wrong about this being the best performing way since the headers could be any `Seq` implementation), did it like this because I remembered how rendering headers was a big expensive post in the http benchmarks from the profiling we did last summer.